pgdrive/scene_creator/vehicle_module/routing_localization.py

```diff
@@ -81,7 +81,15 @@ def update(self, map: Map, start_road_node=None, final_road_node=None, random_se
             start_road_node = FirstBlock.NODE_1
         if final_road_node is None:
             random_seed = random_seed if random_seed is not False else map.random_seed
-            socket = get_np_random(random_seed).choice(map.blocks[-1].get_socket_list())
+            sockets = map.blocks[-1].get_socket_list()
+            while True:
+                socket = get_np_random(random_seed).choice(sockets)
+                if not socket.is_socket_node(start_road_node):
+                    break
+                else:
+                    sockets.remove(socket)
+                    if len(sockets) == 0:
+                        raise ValueError("Can not set a destination!")
```

Comment:
In which case this error would happen? Can we make sure that case will never happens? (If so then close this comment)